integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/noarkexchange/altinn/AltinnConnectionCheck.java

```diff
@@ -29,15 +29,19 @@ public void checkTheConnection() {
         } catch (IBrokerServiceExternalBasicCheckIfAvailableFilesBasicAltinnFaultFaultFaultMessage e) {
             throw new MeldingsUtvekslingRuntimeException("Could not check for available files from Altinn: " + AltinnReasonFactory.from(e), e);
         }
-        //Verifying integrity of DPO delegation of the on-behalf-of-reportees
+
+        verifyIntegrityOfDPODelegationForOnBehalfOfReportees();
+    }
+
+    private void verifyIntegrityOfDPODelegationForOnBehalfOfReportees() {
         Set<String> reportees = props.getDpo().getReportees();
         if (reportees != null && !reportees.isEmpty()) {
             for (String reportee : reportees) {
                 try {
                     altinnWsClient.checkIfAvailableFiles(reportee);
                     log.info("Altinn DPO delegation verified for reportee: {}", reportee);
                 } catch (IBrokerServiceExternalBasicCheckIfAvailableFilesBasicAltinnFaultFaultFaultMessage e) {
-                    log.warn("Failed to verify DPO delegation for reportee: {}", reportee, e);
+                    throw new RuntimeException("Failed to verify DPO delegation for reportee: {}" +  reportee + e);
```

Comment:
Anbefalar å bruka ein annan RuntimeException-konstruktør, den som tek inn ein streng og ein Throwable-instans, og gi inn IBrokerServiceExternalBasicCheckIfAvailableFilesBasicAltinnFaultFaultFaultMessage-instansen som sistnemnde parameter. Då wrappar du Altinn-exceptionen, i staden for å oversetja det til ein potensielt relativt intetsigande streng som vert slegen saman med "Failed to verify...".

Reply:
Fikset